Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java

```diff
@@ -445,8 +445,12 @@ void setProcess(Process process) {
         this.process = process;
     }
 
-    void switchStructure(Object treeNodeData) throws InvalidMetadataValueException, NoSuchMetadataFieldException {
-        metadataPanel.preserveLogical();
+    void switchStructure(Object treeNodeData) throws NoSuchMetadataFieldException {
+        try {
+            metadataPanel.preserveLogical();
+        } catch (InvalidMetadataValueException e) {
+            logger.info(e.getLocalizedMessage());
+        }
         metadataPanel.showLogical(structurePanel.getSelectedStructure());
```

Comment:
I know this part of code is not changed but in case you would have time to change you could assign _structurePanel.getSelectedStructure()_ to variable. It would save us false positive messages that _get()_ is used without _isPresent()_ call.

Reply:
The `isPresent` check is done inside the `showLogical` method. Are you suggesting to change that method so it expects parameters of type `IncludedStructuralElement` instead of `Optional<IncludedStructuralElement>`?

Reply:
Next check is done also in this method:
```
                if (structureTreeNode.getDataObject() instanceof IncludedStructuralElement
                        && structurePanel.getSelectedStructure().isPresent()) {
                    // Logical structure element selected
                    IncludedStructuralElement structuralElement = structurePanel.getSelectedStructure().get();
```

Reply:
Next check is also done in this method:
```
                if (structureTreeNode.getDataObject() instanceof IncludedStructuralElement
                        && structurePanel.getSelectedStructure().isPresent()) {
                    // Logical structure element selected
                    IncludedStructuralElement structuralElement = structurePanel.getSelectedStructure().get();
```